ChopChop/Models/Graph/DirectedAcyclicGraph.swift

```diff
@@ -0,0 +1,215 @@
+/**
+ The `DirectedAcyclicGraph` ADT is able to represent a simple, directed, acyclic graph.
+ 
+ The representation invariants for a DAG g are:
+ - g is a simple graph (multiple edges with the same source and destination node not allowed)
+ - g is a directed graph
+ - g is acyclic (does not contain a cycle)
+ */
+class DirectedAcyclicGraph<N: Node>: Graph<N> {
```

Comment:
just trying to pre empt the comments we r gonna get, is a DAG substitutable for a graph?

Reply:
I think it should be substitutable. None of the `Graph` methods are "illegal" in a DAG